sdk/auth/token_adding_interceptor.go

```diff
@@ -82,6 +83,38 @@ func (i TokenAddingInterceptor) AddCredentials(
 	return err
 }
 
+func (i TokenAddingInterceptor) AddCredentialsConnect() connect.UnaryInterceptorFunc {
+	return connect.UnaryInterceptorFunc(func(next connect.UnaryFunc) connect.UnaryFunc {
+		return func(
+			ctx context.Context,
+			req connect.AnyRequest,
+		) (connect.AnyResponse, error) {
+			accessToken, err := i.tokenSource.AccessToken(ctx, i.httpClient)
+			if err != nil {
+				slog.ErrorContext(ctx, "error getting access token", "error", err)
+				return nil, connect.NewError(connect.CodeUnauthenticated, err)
+			}
+
+			// Add Authorization header
+			req.Header().Set("Authorization", fmt.Sprintf("DPoP %s", accessToken))
+
+			// Add DPoP header if possible
+			dpopTok, err := i.GetDPoPToken(req.Spec().Procedure, http.MethodPost, string(accessToken))
```

Comment:
If Dpop doesn't work, shouldn't line 99 be Bearer? I forget :-/

Reply:
im just mimicking the behavior of the existing grpc interceptor which seems to add "Authorization DPoP ..." https://github.com/opentdf/platform/blob/5ff754e99189d09ec3698128d1bc51b6f7a90994/sdk/auth/token_adding_interceptor.go#L60

Reply:
It sounds like it should be Bearer in this case, as the DPoP spec says a server MUST check for the dpop header when the authorization header creds are in the form DPoP ...

https://datatracker.ietf.org/doc/html/rfc9449#section-7.1

That said, if this isn't a regression I'm fine with you filing a follow up ticket that is about improving our DPoP testing infrastructure. Also this is a pretty bad edge case so IDK what the 'right' thing to do here it - it probably is to fail the request, not to try it with a busted dpop

Reply:
we must be adding the bearer header elsewhere otherwise id assume all the tests would fail at casbin